### DISCUSSION TITLE
有沒有人可以幫我解惑

### DISCUSSION THREAD

**Discussion by Isongzhe** [1101-GS4719-C/Python-Community-GS4719-C#166] [Q&A]
![image](https://user-images.githubusercontent.com/86179263/143185575-0dfea62a-5a44-47e8-9bd2-0921ab4e4b3a.png)
我用講義上的例子，把兩個字典用update匯入到dict1，那他dict1到底會長什麼樣子?(他是把dict2加入到dict1的後面，還是隨機插入)
為什麼用print(dict1) 跟 直接呼叫dict1 兩個的結果會不一樣? 
在我的認知裡，字典應該是沒有所謂的順序，我感覺他會是隨機的插入到dict1裡，那為什麼我每一次的結果都會是固定的?

我是小菜雞，有沒有人可以幫我解答我的疑惑?

**Comment by EricAgumonJou** [comment]
我的認知是dict2會直接加入到dict1的後面，原因的話...我也想問ＸＤ

**Comment by enoch-lightning** [comment]
我認為字典使用.update()函式，只是把原字典沒有，但新字典有的東西，加進原字典。

**Comment by enoch-lightning** [comment]
![image](https://user-images.githubusercontent.com/86505454/143196676-86876f8f-7395-4b1e-a6b1-63547a820bb4.png)
在你寫的while迴圈中，跑第一次迴圈，執行完`dict1.update(dict2)`時，已經將dict1中沒有，但dict2中有的key:value配對加到dict1中。跑第二次(x=1)、第三次(x=2)、第四次(x=3)與第五次(x=4)迴圈時，`dict1.update(dict2)`不會再變更dict1的值，因為dict2中有的key:value配對在跑第一次(x=0)迴圈時，就已經增加(更新)到dict1裡。所以，我不太了解你將dict1.update(dict2)放在while迴圈裡的用意。

**Reply by Isongzhe** [reply]
迴圈有錯的話很抱歉，但我原本只是想試看看，重複update 的話，他的dict1的樣子會變嗎?  因為我不太理解，如果字典是沒有順序的話，那插入新字典的排列方式不會是隨機的嗎?

**Reply by enoch-lightning** [reply]
上面是我的想法，也不一定完全正確。我會問你寫迴圈的用意，就是想了解你想做什麼。我來想想看是否能用程式驗證你的問題。

**Reply by enoch-lightning** [reply]
我發現自己也有與你相同的疑惑。

**Comment by ghost** [comment]
我沒有執行出你37方格的結果，我是用print(dict1)跟直接打dict都是一樣的結果，也許是python版本不同的關係
關於順序的部分我有一些想法，雖然字典是non-sequence，但應該會記錄你加入的順序，因為字典有一個方法是.popitem()，請參考圖片，也許可以算是一種解釋?
![1638089834237](https://user-images.githubusercontent.com/90567166/143737421-7ba382ae-4ef1-4006-a981-18b22770698d.jpg)
